### ISSUE TITLE
Node pool's version field does not use default_auto_upgrade value

### ISSUE THREAD

**Issue by samarthd** [terraform-google-modules/terraform-google-kubernetes-engine#1182]
### TL;DR

The `management` block in the `google_container_node_pool` calculates `auto_upgrade` like so: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/774fc12df6cb9f801caf72cd0ac6dfdbe723da46/cluster.tf#L261

The `version` field relies on `auto_upgrade`, but defaults to `false` instead of `local.default_auto_upgrade`:
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/774fc12df6cb9f801caf72cd0ac6dfdbe723da46/cluster.tf#L234-L238

This means that when `auto_upgrade` is not explicitly specified and [the cluster is regional](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/774fc12df6cb9f801caf72cd0ac6dfdbe723da46/main.tf#L69-L70), the node pool's `version` field will disagree with the `management.auto_upgrade` setting.

### Expected behavior

Node pool version should use `local.default_auto_upgrade` as the default for `auto_upgrade` if the value isn't specified

### Observed behavior

Node pool version uses `false` as the default for `auto_upgrade` if the value isn't specified

### Terraform Configuration

```hcl
N/A
```

### Terraform Version

```sh
N/A
```
